packages/semi-ui/_portal/index.tsx

```diff
@@ -31,46 +52,42 @@ class Portal extends PureComponent<PortalProps, PortalState> {
     static propTypes = {
         children: PropTypes.node,
         prefixCls: PropTypes.string,
-        getPopupContainer: PropTypes.func,
+        style: PropTypes.object,
+        initStyle: PropTypes.object,
         className: PropTypes.string,
+
+        getPopupContainer: PropTypes.func,
         didUpdate: PropTypes.func,
     };
 
     el: HTMLElement;
     context: ContextValue;
+
     constructor(props: PortalProps, context: ContextValue) {
         super(props);
+        /**
+         * Unlike other implementations, we initize `container` and `el` immediately to get children's refs in this tick.
+         * See issue: https://github.com/DouyinFE/semi-design/issues/1703
+         * 
+         * We only append `el` to `container` at `componentDidMount` to fix react18 strict mode error.
+         */
         this.state = {
-            container: this.initContainer(context, true)
+            container: this.getContainer(context)
```

Comment:
Is there an example or test case to explain this as I always get error running nextjs with semi@^2.47.0

https://codesandbox.io/p/devbox/holy-wood-pntvs5?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clp9d5naj000c3b6hfwtnmi8t%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clp9d5naj00033b6hzvwru8p6%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clp9d5naj00093b6htj4j1sel%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clp9d5naj000b3b6htx17chf8%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clp9d5naj00033b6hzvwru8p6%2522%253A%257B%2522id%2522%253A%2522clp9d5naj00033b6hzvwru8p6%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clp9d5naj000b3b6htx17chf8%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clp9d5naj000a3b6hdi6ikjya%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clp9d5naj000b3b6htx17chf8%2522%252C%2522activeTabId%2522%253A%2522clp9d5naj000a3b6hdi6ikjya%2522%257D%252C%2522clp9d5naj00093b6htj4j1sel%2522%253A%257B%2522id%2522%253A%2522clp9d5naj00093b6htj4j1sel%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clp9d5naj00043b6hrhtjhv72%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clp9d5naj00053b6hh8blaqqf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522build%2522%257D%252C%257B%2522id%2522%253A%2522clp9d5naj00063b6hexza3pew%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%252C%257B%2522id%2522%253A%2522clp9d5naj00073b6ha6p99416%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522lint%2522%257D%252C%257B%2522id%2522%253A%2522clp9d5naj00083b6h1fb2neeh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522install%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clp9d5naj00043b6hrhtjhv72%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

Reply:
Understood. After researching this issue, I found no definitive solution. The most recommended approaches are using dynamic import or returning null when the document doesn't exist. I will experiment with the latter option. However, please be aware that this may trigger React errors [418](https://legacy.reactjs.org/docs/error-decoder.html/?invariant=418) and [423](https://legacy.reactjs.org/docs/error-decoder.html/?invariant=423).